drivers/clock_control/nrf_power_clock.c

```diff
@@ -16,6 +17,20 @@ LOG_MODULE_REGISTER(clock_control, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
 
 #define DT_DRV_COMPAT nordic_nrf_clock
 
+
+#define CTX_ONOFF		BIT(6)
+#define CTX_API			BIT(7)
+#define CTX_MASK (CTX_ONOFF | CTX_API)
+
+/* Used only by HF clock */
+#define HF_BT_ACTIVE		BIT(8)
+#define HF_GENERIC_ACTIVE	BIT(9)
+#define HF_USER_MASK		(HF_BT_ACTIVE | HF_GENERIC_ACTIVE)
```

Comment:
Since this is only for HFCLK, why not use a dedicated variable for it? Something like:
```
static atomic_t hfclk_users;
```
You would then avoid the need of this weird masking of `flags` with `HF_USER_MASK` in `set_off_state()`:
https://github.com/zephyrproject-rtos/zephyr/blob/a7fbf5e80f3f702024c261dd63899f368f39028d/drivers/clock_control/nrf_power_clock.c#L158
and in `set_starting_state()`:
https://github.com/zephyrproject-rtos/zephyr/blob/a7fbf5e80f3f702024c261dd63899f368f39028d/drivers/clock_control/nrf_power_clock.c#L173-L175
Also accesses to such variable should be faster than to `flags` for which you need to get the pointer via `get_hf_flags()`.